File: frontend/lib/models/Post.dart
class Post {
  final int postId;
  final String voiceUrl;
  final double voiceLength;
  final String address;
  final String text;
  final List<String> photoList;
  bool isFavorite;
  int favoriteCount;
  final bool isMine;
  bool isExpand;

  Post({
    required this.postId,
    required this.voiceUrl,
    required this.voiceLength,
    required this.address,
    required this.text,
    required this.photoList,
    required this.isFavorite,
    required this.favoriteCount,
    required this.isMine,
    this.isExpand = false,
  });

  // Deserialize(파싱?)
  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      postId: json['postId'],
      voiceUrl: json['voiceUrl'],
      voiceLength: json['voiceLength'],
      address: (json['address2'] ?? '') +
          ' ' +
          (json['address3'] ?? '') +
          ' ' +
          (json['address4'] ?? ''),
      text: json['text'],
      photoList: List.from(json['photoList']),
      isFavorite: json['isFavorite'],
      favoriteCount: json['favoriteCount'],
      isMine: json['isMine'],
    );
  }
}

File: backend1/src/main/java/com/danim/entity/Post.java
package com.danim.entity;

import lombok.*;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
@DynamicInsert
public class Post extends BaseTime{
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	@Column(name = "post_id", nullable = false)
	private Long postId;
	
	private String voiceUrl;
	
	private Double voiceLength;
	
	private String nationUrl;
	
	private String address1;
	private String address2;
	private String address3;
	private String address4;

	private String text;

	// Post 테이블과 TimeLine 테이블 FK
	@ManyToOne
	@JoinColumn(name="timeline_id")
	@ToString.Exclude
	@OnDelete(action = OnDeleteAction.CASCADE)
	private TimeLine timelineId;

	// Post 테이블과 Nation 테이블 FK
	@ManyToOne
	@JoinColumn(name="nation_id")
	@ToString.Exclude

	private Nation nationId;

	@OneToMany(mappedBy = "postId",fetch = FetchType.EAGER)
	@Builder.Default
	@OnDelete(action = OnDeleteAction.CASCADE)
	private List<Photo> photoList = new ArrayList<>(); // photoId 리스트
}